index.js

```diff
@@ -133,6 +134,11 @@ module.exports = function (schema, options) {
     // Post hook that gets called after any save or update
     // operation and that filters unique errors
     var postHook = function (error, doc, next) {
+
+        if (!doc) {
+            doc = this;
+        }
```

Comment:
If we cannot rely on the document being passed as a second argument, we should maybe make that clear by always using `this`:

```js
var postHook = function (error, _, next) {
    var doc = this;
    // …
```